File: CS/ReportParameterMVC/Reports/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace ReportParameterMVC.Reports
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();
        }

        private void XtraReport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            this.ordersTableAdapter.FillByOrderID(this.dataSet11.Orders, (int)this.OrderID.Value);
        }

    }
}

File: VB/ReportParameterMVC/Reports/XtraReport1.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace ReportParameterMVC.Reports

    Public Partial Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub XtraReport1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            ordersTableAdapter.FillByOrderID(dataSet11.Orders, CInt(OrderID.Value))
        End Sub
    End Class
End Namespace
